### PR TITLE
feat(Text): Add option to turn off baseline for text

### PR DESCRIPTION
**Pull request by dborysiewicz**
Added option for type mixins and `Text` component to support turning off the baseline modifier (basekick) this will allow use to vertical centre text.

Not I'm sure if I'm happy with the idea of `.nonBaselineText()` being referenced in `baselineType.less`? We could just not use a mixin and move the code directly or maybe just use `.rawText()` - thoughts?

### CODE REVIEW

File: react/Text/Text.demo.js [hunk 1]
```diff
@@ -19,6 +19,13 @@ export default {
             ...props,
             strong: true
           })
+        },
+        {
+          label: 'Baseline False',
```
Comment (markdalgleish) [inline react/Text/Text.demo.js]: Nitpick: Maybe "Disable baseline" instead?

File: theme/type/baselineType.less [hunk 2]
```diff
@@ -2,11 +2,16 @@
...
+.__baselineType(@type-scale; @row-span; @baseline) when (isnumber(@type-scale)) and (isnumber(@row-span)) and (@baseline = true) {
   .basekick(@type-scale,
             @row-span,
             @font-descender-height-scale,
             @base-font-size,
             @grid-row-height);
 }
+
+.__baselineType(@type-scale; @row-span; @baseline) when (isnumber(@type-scale)) and (isnumber(@row-span)) and (@baseline = false) {
+  .nonBaselineText(@type-scale, @row-span);
```
Comment (markdalgleish) [inline theme/type/baselineType.less]: Should we just inline this mixin, since it's not used anywhere else?

File: theme/type/baselineType.less [hunk 3]
```diff
@@ -2,11 +2,16 @@
 
 @import (reference) "../layout/grid";
 @import (reference) "type";
+@import (reference) "../mixins/type";
 
-.__baselineType(@type-scale; @row-span) when (isnumber(@type-scale)) and (isnumber(@row-span)) {
+.__baselineType(@type-scale; @row-span; @baseline) when (isnumber(@type-scale)) and (isnumber(@row-span)) and (@baseline = true) {
```
Comment (markdalgleish) [inline theme/type/baselineType.less]: I wonder if this file and its mixins needs to be renamed now that it's not always about being on baseline? 🤔

Comment (dborysiewicz) [inline theme/type/baselineType.less]: Yep this and your other comment is what I was getting at not 100% happy... I'll chat to you on Slack.

Comment (markdalgleish) [inline theme/type/baselineType.less]: lol, don't mind me, just stealing your comments and palming them off as my own 💪

File: react/Text/Text.less [hunk 4]
```diff
@@ -20,11 +19,19 @@
   }
 }
 
-.small { .smallTextResponsive(); }
-.subheading { .subheadingTextResponsive(); }
-.headline { .headlineTextResponsive(); }
-.heading { .headingTextResponsive(); }
-.hero { .heroTextResponsive(); }
+.root:not(.baseline) { .standardTextResponsive(@baseline: false); }
```
Comment (georgespyropoulos) [inline react/Text/Text.less]: What about grouping these by type rathen than by baseline / non-base line eg.

```
.root {
  &:not(.baseline) {.standardTextResponsive(@baseline: false); }
  &.baseline { .standardTextResponsive(@baseline: true); }
}
```

**Comment by mengtzu** [comment]
👍

**Comment by kosanna** [comment]
Ah yes, was actually going to ask how we do vertical centering having baseline. Cool!

**Pull request merged**